File: lingting-huawei/src/main/java/live/lingting/framework/huawei/HuaweiObsObject.java
package live.lingting.framework.huawei;

import live.lingting.framework.aws.AwsS3Object;
import live.lingting.framework.aws.s3.interfaces.AwsS3ObjectDelegation;
import live.lingting.framework.http.header.HttpHeaders;
import live.lingting.framework.huawei.obs.HuaweiObsHeaders;
import live.lingting.framework.huawei.properties.HuaweiObsProperties;

/**
 * @author lingting 2024-09-13 14:48
 */
public class HuaweiObsObject extends HuaweiObs<AwsS3Object> implements AwsS3ObjectDelegation {

	public HuaweiObsObject(HuaweiObsProperties properties, String key) {
		super(new AwsS3Object(properties, key));
	}

	@Override
	public HuaweiObsHeaders head() {
		HttpHeaders head = AwsS3ObjectDelegation.super.head();
		return new HuaweiObsHeaders(head);
	}

}

File: lingting-huawei/src/main/kotlin/live/lingting/framework/huawei/HuaweiObsObject.kt
package live.lingting.framework.huawei

import live.lingting.framework.aws.AwsS3Object
import live.lingting.framework.aws.s3.interfaces.AwsS3ObjectDelegation
import live.lingting.framework.huawei.obs.HuaweiObsMeta
import live.lingting.framework.huawei.properties.HuaweiObsProperties

/**
 * @author lingting 2024-09-13 14:48
 */
class HuaweiObsObject(properties: HuaweiObsProperties, key: String) : HuaweiObs<AwsS3Object>(AwsS3Object(properties, key)), AwsS3ObjectDelegation {

    override fun head(): HuaweiObsMeta {
        val head = super<AwsS3ObjectDelegation>.head()
        return HuaweiObsMeta(head)
    }

}
